docs/assignment3.md

```diff
@@ -0,0 +1,503 @@
+
+<b> Game Class - State Machine Diagram </b>
+
+The state machine diagrams, same as the whole design and the implementation has been through quite some changes. First of all, within a previous version, a very common design mistake has been made whereas the state machine diagrams became activity diagrams, due to the naming of states and of the events. This has been updated upon feedback and the diagram has become as follows (for the previous version).
+
+![State Machine Diagram - Game](images/gamestate2.jpeg)
+
+
+Within this diagram, it can be seen that the **game** class has four finite states which are “Game initialized”, “Game starting”, “Command reading” and “Command handling”. The initialization process within this state machine is represented by a simple state, and the other states are triggered through activities and events. The “Game initialized” is triggered by “start listener”, the “Command reading” is triggered by “read command string”, in both cases as, and the “Command Handling” has to trigger within this diagram.
+
+Not only the naming had to be further adjusted, but the initialization and the triggers changed as the whole design changed. This led to a new state machine diagram for the class **game** as it can be found below.
+
+![Final State Machine Diagram – Game ](images/Gamestateass3.png)
+
+As it can be seen, the names of the states have been more appropriately named here in order to point out that these are states, and the names are nouns. 
+
+The “Game initialized” state, is no longer a simple state but it became a composite state, whereas the internal activities performed within this state are as follows: it initializes an instance of all needed classes, initializes the stack for the views, set the first and current scene id and initializes the listener. 
+
+The next state is triggered by the activity to get all these variables. This state is called “Game started”. When entering this scene, all the variables are received. Then, as internal activities, this state adds a client to the environment, runs the listener and makes sure that a view can be entered. 
+
+The next state is triggered by the event of reading input from the terminal, and then the activity of checking the string that has been read. Then the system enters the state “Command Read”. Here, the state receives the input, and then checks and verifies whether it is a valid input. 
+
+The next state is triggered by the handle command activity. Here the command is read, and as an internal activity the state enters the scene corresponding to that command. Through here, the system will go through a number of states that belong to the **scene** class. Then, at exiting those states, the “Command Handling” exits with a new scene. The listener is still active throughout this class, and if it is triggered by new input, then the “Command Read” state will be reached again. Therefore the **game** class can reach these two states in a continuous loop. 
+
```

Comment:
You mention that "Game initialized" is now a composite state, where are it's substates? Good otherwise.